rotkehlchen/rotkehlchen.py

```diff
@@ -310,7 +310,9 @@ def unlock_user(
         Inquirer()._manualcurrent.set_database(database=self.data.db)
 
         # Initialize the cached settings singleton
-        CachedSettings()
+        with self.data.db.conn.read_ctx() as cursor:
+            settings = self.get_settings(cursor)
+        CachedSettings().update_entries(settings)
```

Comment:
instead of opening two cursors move the update of the cached settings to where the settings were queried before